vmw_backdoor can't be cross-compiled on a non-x86_64 host

First of all, thanks for making this crate! It's fantastic. We're using it in our own cloud-init replacement in Bottlerocket.

We ran into https://github.com/bottlerocket-os/bottlerocket/issues/1484 today and I wanted to get your thoughts on a potential fix.

The build script has this conditional:
```
fn main() {
    #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
    asm_x86_64_linux();
}
```

However, since build scripts are compiled for the host, this effectively turns the script into a no-op when built on a non-x86_64 architecture. When the build script is run later, even if `--target` and `TARGET_CC` are set correctly, the script will not compile the x86_64 assembly. Crates that make use of those symbols will then fail to link properly.

To fix cross-compiling, it's sufficient to remove that `cfg` directive from the build script. I'm not entirely sure what impact this would have on downstream crates. For the most part, I expect they would need to use similar conditionals to avoid the crate dependency and to prevent calling into the functions backed by the assembly.

Yup, that works for me, thanks!

Here's a [test case](https://gist.github.com/bcressey/8f7ae28d34f8e39045a03cc9cc543d50) if you have access to an arm64 system. It's not particularly minimal because it requires a glibc build and install.

There's a different problem where musl builds fail for this reason:
```
error[E0425]: cannot find function `iopl` in crate `libc`
   --> /tmp/vmw_backdoor-rs/src/backdoor.rs:101:34
    |
101 |         let err = unsafe { libc::iopl(level) };
    |                                  ^^^^ not found in `libc`
```

That's more of a deficiency in Rust's libc crate, though; `iopl` seems to be [special-cased for x86_64 gnu](https://github.com/rust-lang/libc/blob/master/src/unix/linux_like/linux/gnu/b64/x86_64/mod.rs#L846).

If you wanted to guard against that, you could also check that `CARGO_CFG_TARGET_ENV` was "gnu". But then you'd have to revisit the build script if `libc` changed, which doesn't seem ideal. It would also obscure the actual problem - musl builds would fail with the symbol errors at link time after the assembly was quietly skipped. The compile error seems much clearer.

--- a/build.rs
+++ b/build.rs
@@ -1,8 +1,13 @@
 #![allow(dead_code)]
 
 fn main() {
-    #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-    asm_x86_64_linux();
+    let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH")
+        .expect("missing $CARGO_CFG_TARGET_ARCH env variable");
+    let target_os =
+        std::env::var("CARGO_CFG_TARGET_OS").expect("missing $CARGO_CFG_TARGET_OS env variable");
+    if target_arch == "x86_64" && target_os == "linux" {
+        asm_x86_64_linux();
+    };
 }
 
 fn asm_x86_64_linux() {

